src/tasks/onnx.py

```diff
@@ -0,0 +1,107 @@
+from typing import Dict, Union, Any
+
+import onnx
+import torch
+import numpy as np
+from torch import Tensor
+import onnxruntime as onnxrt
+from omegaconf import DictConfig
+
+from src.constructor import TASKS
+from src.tasks.base import BaseTask
+from src.constructor.config_structure import Phase
+
+
+@TASKS.register_class
+class ONNXTask(BaseTask):
+    """A class for onnx task."""
+
+    str_type2numpy_type = {'tensor(float)': 'float32', 'tensor(float16)': 'float16',
+                           'tensor(double)': 'float64', 'tensor(int8)': 'int8',
+                           'tensor(int16)': 'int16', 'tensor(int32)': 'int32',
+                           'tensor(int64)': 'int64', 'tensor(uint8)': 'uint8'}
+
+    def __init__(self, hparams: DictConfig):
+        """Init ONNXTask.
+
+        Args:
+            hparams: Hyperparameters that set in yaml file.
+        """
+        super().__init__(hparams)
+        self.__infer_params = self._hparams.task.params
+        model_path = self.__infer_params.path_to_onnx
+
+        onnx_model = onnx.load(model_path)
+        onnx.checker.check_model(onnx_model)
+
+        self._sess = onnxrt.InferenceSession(model_path, providers=self.__infer_params.providers)
+        self.__binding = self._sess.io_binding()
+
+        self.__inputs = [{'name': item.name,
+                          'dtype': self.str_type2numpy_type[item.type]}
+                          for item in self._sess.get_inputs()]
+
+        self.onnx2dataset = self._hparams.task.params.keys_mapping_onnx2dataset
```

Comment:
why it not private? I think it's better to change name to keys_mapping_onnx2dataset like in yaml